### PR TITLE
Add Resource binding test

### PR DESCRIPTION
**Pull request by guusdk**
A new test that asserts that the server closes the connection with an error, when a stanza is sent prior to resource binding (as defined by RFC 3620, section 7.1).

Smack doesn't offer an API that can be utilized to send a stanza between authentication and resource binding. For that reason, this test sends stanzas prior to authentiation (instead of post auth, pre bind).

### CODE REVIEW

**Comment by guusdk** [comment]
This tests for a bug that is fixed in Openfire 4.8.0: https://igniterealtime.atlassian.net/jira/software/c/projects/OF/issues/OF-2565

This PR depends on the following (or similar) changes in Smack:
- https://github.com/igniterealtime/Smack/pull/542 (to make sure that the test is executed on all connection types, instead of repeated on the same connection)
- https://github.com/igniterealtime/Smack/pull/543 (to allow this test to run against Openfire that uses its default configuration for the port to be used for websockets)

**Comment by guusdk** [comment]
Rebased.

The current build failure is caused by Smack's SNAPSHOT artifacts in maven-central to be compiled with Java 11, which is not backwards compatible with Java 8.

@Flowdalic (assuming that the required changes won't be ported to Smack 4.4.x) will Java 8-support be dropped in Smack 4.5, or is the uploaded artifact incorrect?

**Comment by guusdk** [comment]
As Smack's integration-test module now uses Java 11, this project should follow. I've created https://github.com/XMPP-Interop-Testing/smack-sint-server-extensions/pull/11 to upgrade from Java 8 to 11.

This PR should be rebased after #11 gets merged.

**Comment by guusdk** [comment]
#11 is merged. Rebased this PR onto main.

File: src/main/java/org/jivesoftware/smack/ResourceBindingIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Tests that verify the behavior of Openfire around resource binding.
+ *
+ * @author Guus der Kinderen, guus.der.kinderen@gmail.com
+ */
+@SpecificationReference(document = "RFC 6120")
```
Comment (Fishbowler) [inline src/main/java/org/jivesoftware/smack/ResourceBindingIntegrationTest.java:44]:
```suggestion
@SpecificationReference(document = "RFC6120")
```
Remove space to match the junit `specifications.properties`

**Review by Fishbowler — APPROVED** [review]

File: src/main/java/org/jivesoftware/smack/ResourceBindingIntegrationTest.java (hunk 1 above)
Comment (Fishbowler) [reply src/main/java/org/jivesoftware/smack/ResourceBindingIntegrationTest.java:44]: ...or we can change the junit references

Comment (Fishbowler) [reply src/main/java/org/jivesoftware/smack/ResourceBindingIntegrationTest.java:44]: ...or we can change the junit references

**Pull request merged**